Lib/pyclbr.py

```diff
@@ -174,17 +198,22 @@ def _readmodule(module, path, inpackage=None):
                 tokentype, meth_name, start = next(g)[0:3]
                 if tokentype != NAME:
                     continue # Syntax error
+                cur_func = None
                 if stack:
-                    cur_class = stack[-1][0]
-                    if isinstance(cur_class, Class):
-                        # it's a method
-                        cur_class._addmethod(meth_name, lineno)
-                    # else it's a nested def
+                    cur_obj = stack[-1][0]
+                    if isinstance(cur_obj, Object):
```

Comment:
Yes, the stack can be empty.  That is why accessing the last object on the stack is protected by 'if stack'.  

If  the stack is not empty, *and* and the isinstance check fails, then cur_func will not be rebound to a new Function object corresponding to to the new def.  So, as I tried to say, either the current code is buggy or the isinstance check (which is changed by the patch) is no longer needed.  In the old code, nested functions, including methods were never stacked, but methods were added to the separate methods attribute of class objects.    I will come back to this after checking the other changes.